### PR TITLE
Introduce Monaco JSON editor

### PR DESCRIPTION
**Pull request by swordEdge**

### CODE REVIEW

File: src/AsyncMessageChannel.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export class AsyncMessageChannel {
 
   public connect() {
     return this.attachMessageListener(async (msg: { id?: string; message?: AsyncMessages }) => {
-      if (!msg.id || !msg.message || !msg.message.type.startsWith('async/')) return;
+      if (!msg?.id || !msg.message || !msg.message.type.startsWith('async/')) return;
```
Comment (six7) [inline src/AsyncMessageChannel.ts:66]: shouldn't `msg` always be present?

File: src/app/components/JSONEditor.tsx [hunk 2]
```diff
@@ -1,24 +1,39 @@
+/*
```
Comment (six7) [inline src/app/components/JSONEditor.tsx:1]: we can remove those comments

File: src/app/components/JSONEditor.tsx [hunk 3]
```diff
@@ -1,24 +1,39 @@
...
   handleChange: (tokens: string) => void;
-  hasError: boolean;
 };
 
 function JSONEditor({
   stringTokens,
   handleChange,
-  hasError,
 }: Props) {
-  const editProhibited = useSelector(editProhibitedSelector);
```
Comment (six7) [inline src/app/components/JSONEditor.tsx:20]: we will still need `editProhibited` for the case where users are in read-only mode

File: src/app/components/JSONEditor.tsx [hunk 4]
```diff
@@ -1,24 +1,39 @@
...
   const { handleJSONUpdate } = useTokens();
+
+  const handleJsonEditChange = React.useCallback((value?: string) => {
+    if (value) handleChange(value);
```
Comment (six7) [inline src/app/components/JSONEditor.tsx:34]: is there a case where value is null? shouldnt the editor *always* have a value? even when its `""` ?

File: src/app/components/JSONEditor.tsx [hunk 5]
```diff
@@ -37,13 +52,11 @@ function JSONEditor({
...
-        isDisabled={editProhibited}
-        placeholder="Enter JSON"
-        rows={21}
-        onChange={handleChange}
+      <Editor
+        language="json"
+        onChange={handleJsonEditChange}
         value={stringTokens}
-        css={{ paddingBottom: hasError ? '$9' : '0' }}
+        theme="vs-dark"
```
Comment (six7) [inline src/app/components/JSONEditor.tsx:59]:
```suggestion
        theme="vs-dark"
        options={{
          minimap: {
            enabled: false,
          },
          lineNumbers: 'off',
          wordWrap: 'on',
          contextmenu: false,
        }}
```

**Review by six7 — CHANGES_REQUESTED** [review]

File: src/app/components/JSONEditor.tsx (hunk 4 above)
Comment (swordEdge) [reply src/app/components/JSONEditor.tsx:34]: The type of parameter of onChange function in Moncaco editor is string | undefined.

**Review by six7 — APPROVED** [review]
🙌

**Pull request merged**